ibtk/src/math/PETScMatUtilities.cpp

```diff
@@ -806,10 +596,731 @@ void PETScMatUtilities::constructPatchLevelSCInterpOp(Mat& mat,
+#endif
+                            box_local_dofs.push_back(dof_idx);
+                        }
+                    }
+                    const int n_idx = static_cast<int>(box_local_dofs.size());
+                    ISCreateGeneral(PETSC_COMM_SELF, n_idx, &box_local_dofs[0], PETSC_COPY_VALUES,
+                                    &is_nonoverlap[0][counter]);
+
+                    // Fill IS'es for overlap subdomain
+                    if (!there_is_overlap)
+                    {
+                        PetscObjectReference((PetscObject)is_nonoverlap[0][counter]);
+                        is_overlap[0][counter] = is_nonoverlap[0][counter];
+                    }
+                    else
+                    {
+                        // Create a box to iterate over overlapping subdomain
+                        const int gcwx = overlap_size(0);
+                        const int gcwy = overlap_size(1);
+                        const int gcwz = overlap_size(2);
+                        IntVector<NDIM> box_lower(i_lower - gcwx, j_lower - gcwy, k_lower - gcwz);
+                        IntVector<NDIM> box_upper(i_lower + width + gcwx - 1, j_lower + height + gcwy - 1,
+                                                  k_lower + depth + gcwz - 1);
+                        Box<NDIM> box_overlap(box_lower, box_upper);
+
+                        std::vector<int> box_overlap_dofs;
+                        const int max_size = (height + 2 * gcwx) * (width + 2 * gcwy) * (depth + 2 * gcwz) * data_depth;
+                        box_overlap_dofs.reserve(max_size);
+                        for (Box<NDIM>::Iterator b(box_overlap); b; b++)
+                        {
+                            const CellIndex<NDIM>& i = b();
+                            for (int d = 0; d < data_depth; ++d)
+                            {
+                                const int dof_idx = (*dof_data)(i, d);
+
+                                // If keep only the DOFs that are inside the
+                                // physical domain. Some of the DOFs will be on
+                                // other processors.
+                                if (dof_idx >= 0)
+                                {
+                                    box_local_dofs.push_back(dof_idx);
+                                }
+                            }
+                        }
+                        const int n_idx = static_cast<int>(box_overlap_dofs.size());
+                        ISCreateGeneral(PETSC_COMM_SELF, n_idx, &box_overlap_dofs[0], PETSC_COPY_VALUES,
+                                        &is_overlap[0][counter]);
+                    }
+                    ++counter;
+                    i_lower += width;
+                }
+                j_lower += height;
+            }
+            k_lower += depth;
+        }
+#endif
+    }
+
+#if !defined(NDEBUG)
+    TBOX_ASSERT(counter == n_subdomains);
+#endif
+    return;
+} // constructPatchLevelASMSubdomains_cell
+
```

Comment:
yeah. So, if they are, then the above code should do the right thing for c-f and physical boundaries?